### PR TITLE
enhancement/tests

### PR DESCRIPTION
**Pull request by decanus**

### CODE REVIEW

File: contracts/Root.sol [hunk 1]
```diff
@@ -10,8 +10,8 @@ contract Root is Ownable {
 
     using BytesUtils for bytes;
 
-    address public constant DEFAULT_REGISTRAR = 0x0; // @todo, also should we assume this to be a constant?
-    bytes32 public constant ROOT_NODE = keccak256(bytes32(0));
+    address public constant DEFAULT_REGISTRAR = 0x123; // @todo, also should we assume this to be a constant?
```
Comment (Arachnid) [inline contracts/Root.sol]: No, we should probably make  it updatable by the root keyholders.

File: test/TestRoot.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        await ens.setSubnodeOwner(0, web3.sha3('eth'), root.address, {from: accounts[0]});
+        await ens.setOwner(0, root.address);
+    });
+
+    describe('setSubnodeOwner', async () => {
+
+        it('should fail when trying to set subnode owner for non root domain', async () => {
+            try {
+                await root.setSubnodeOwner(web3.sha3('eth'), '0x123', accounts[1], {from: accounts[0]});
+            } catch (error) {
```
Comment (Arachnid) [inline test/TestRoot.js]: Won't this silently succeed if no error occurs? See the `expectFailure` function in, eg, dnssec-oracle for an example of how to handle this easily.

File: test/TestRoot.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+        await ens.setSubnodeOwner(0, web3.sha3('eth'), root.address, {from: accounts[0]});
+        await ens.setOwner(0, root.address);
+    });
+
+    describe('setSubnodeOwner', async () => {
+
+        it('should fail when trying to set subnode owner for non root domain', async () => {
+            try {
+                await root.setSubnodeOwner(web3.sha3('eth'), '0x123', accounts[1], {from: accounts[0]});
```
Comment (Arachnid) [inline test/TestRoot.js]: Why do we  even have the 'node' parameter to this function?

File: test/TestRoot.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+            );
+
+            await root.registerTLD(dns.hexEncodeName('test.'), proof);
+
+            assert.equal(await ens.owner(namehash.hash('test')), accounts[0]);
+        });
+
+        it('should set TLD owner to default registrar when 0x0 is provided', async () => {
+
+            let proof = dns.hexEncodeTXT({
```
Comment (Arachnid) [inline test/TestRoot.js]: This doesn't seem like a thing we should do. If there's *no* text record, it should set it to the default, but if there's any kind of matching text record, we should set it to that.

File: test/TestRoot.js (hunk 2 above)
Comment (decanus) [reply test/TestRoot.js]: yeah I am missing an assert.fail

File: test/TestRoot.js (hunk 3 above)
Comment (decanus) [reply test/TestRoot.js]: fair

File: test/TestRoot.js (hunk 4 above)
Comment (decanus) [reply test/TestRoot.js]: Even for a 0x0 record?

Comment (decanus) [reply test/TestRoot.js]: I don't think we should allow for that out of security tbh

Comment (Arachnid) [reply test/TestRoot.js]: If a domain wants to set the registrar address to 0, meaning nobody can register anything, I think that's fine.

Comment (decanus) [reply test/TestRoot.js]: Ok, I'll need to think about how to do it in the new code that shares the library.

**Pull request merged**